h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -897,38 +894,20 @@ private final double lmax(double[] grad) {
 
 
     private void doCleanup() {
-      try {
-        if(_parms._lambda_search && _parms._is_cv_model)
-          Scope.untrack(removeLater(_dinfo.getWeightsVec()._key));
-        if (_parms._HGLM) {
-          Key[] vecKeys = _toRemove;
-          for (int index=0; index < vecKeys.length; index++) {
-            Vec tempVec = DKV.getGet(vecKeys[index]);
-            tempVec.remove();
-          }
+      if(_parms._lambda_search && _parms._is_cv_model)
+        Scope.untrack(removeLater(_dinfo.getWeightsVec()._key));
+      if (_parms._HGLM) {
+        Key[] vecKeys = _toRemove;
+        for (int index=0; index < vecKeys.length; index++) {
+          Vec tempVec = DKV.getGet(vecKeys[index]);
+          tempVec.remove();
         }
-        if(_model!=null)
```

Comment:
ulocking moved to compute method

Reply:
I think you have a valid point, however, I think this code needs more error handling if this catch is removed, eg.:

Can _dinfo be null? Can the temp vec already be removed from DKV (line 903 possible NPE).

I propose this: 
1) Add the error handling 
2) in onCompletion - invoke doCleanup the regular way
3) in onExceptionalCompletion - we are in an undefined state - any clean-up attempt is by definition best effort only - I would keep the try-catch - and produce a warning in the log

Reply:
sounds good, @sebhrusen can you also take a look?

Reply:
`doCleanUp` is called in `onExceptionalCompletion` (typically if job is aborted), so `_model` is always null in this case.

What would be cleaner probably is to do the unlock only in `onCompletion`, right after calling `doCleanUp`

Reply:
having it moved after compute looks good to me

Reply:
actually no, all other algos do unlock in `computeImpl` and this is necessary for the fix in `computeParams` to work and be able to acquire the lock again